### PR TITLE
Simplifying message API.

### PR DESCRIPTION
**Pull request by Laimiux**
Removing unnecessary message API ceremony. Instead of having multiple message declarations, enable safe arbitrary code execution that will be performed after the state change happens.

**State change with side-effects.**

Old:
```kotlin
transition(state.copy()) { 
  message(input.showToast, "Something happened")
  message(input.closeScreen)
}
```
New:
```kotlin
transition(state.copy()) { 
  input.showToast("Something happened")
  input.closeScreen()
}
```

**Only side-effects (no state change)**

Old:
```kotlin
message(input.showToast, "Something happened")
```
New:
```kotlin
transition { input.showToast("Something happened") }
```

**TODO:**
- [x] Update documentation

### CODE REVIEW

**Comment by carrotkite** [comment]
### JaCoCO Code Coverage 75.34% :white_check_mark:
| Class | Covered | Meta | Status |
|:---|:---:|:---:|:---:|
| `com/instacart/formula/internal/TransitionUtils` | 100% | 0% | :white_check_mark: |
| `com/instacart/formula/internal/FormulaManagerImpl` | 77% | 0% | :white_check_mark: |
| `com/instacart/formula/FormulaRuntime` | 83% | 0% | :white_check_mark: |
| `com/instacart/formula/Transition` | 100% | 0% | :white_check_mark: |
| `com/instacart/formula/test/TestFormulaManager` | 0% | 0% | :white_check_mark: |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: formula/src/main/java/com/instacart/formula/Stream.kt [hunk 1]
```diff
@@ -20,11 +20,11 @@ interface Stream<Message> {
     companion object {
 
         /**
-         * Emits a message when [Stream] is initialized. Use this stream to send a message when [Formula]
+         * Emits a effects when [Stream] is initialized. Use this stream to send a effects when [Formula]
```
Comment (Jawnnypoo) [inline formula/src/main/java/com/instacart/formula/Stream.kt]: `Emits effects` (remove the a). `Use this stream to send effects` (remove the a)

File: formula/src/main/java/com/instacart/formula/Stream.kt [hunk 2]
```diff
@@ -33,12 +33,12 @@ interface Stream<Message> {
         }
 
         /**
-         * Emits a message with [data] when [Stream] is initialized. Uses [data] as key.
+         * Emits a effects with [data] when [Stream] is initialized. Uses [data] as key.
```
Comment (Jawnnypoo) [inline formula/src/main/java/com/instacart/formula/Stream.kt]: Remove a

File: formula/src/main/java/com/instacart/formula/Stream.kt [hunk 3]
```diff
@@ -47,10 +47,10 @@ interface Stream<Message> {
         }
 
         /**
-         * Emits a message when [Formula] is terminated.
+         * Emits a effects when [Formula] is terminated.
```
Comment (Jawnnypoo) [inline formula/src/main/java/com/instacart/formula/Stream.kt]: Remove a

File: formula/src/main/java/com/instacart/formula/Stream.kt [hunk 4]
```diff
@@ -77,7 +77,7 @@ interface Stream<Message> {
 }
 
 /**
- * Emits a message as soon as [Stream] is initialized.
+ * Emits a effects as soon as [Stream] is initialized.
```
Comment (Jawnnypoo) [inline formula/src/main/java/com/instacart/formula/Stream.kt]: Remove a

File: formula/src/test/java/com/instacart/formula/EffectsTimingTest.kt [hunk 5]
```diff
@@ -6,7 +6,7 @@ import com.jakewharton.rxrelay2.PublishRelay
 import io.reactivex.Observable
 import org.junit.Test
 
-class MessageTimingTest {
+class EffectsTimingTest {
 
     @Test fun `state change triggered by a message handler is scoped correctly`() {
```
Comment (Jawnnypoo) [inline formula/src/test/java/com/instacart/formula/EffectsTimingTest.kt]: `state change triggered by an effects handler is scoped correctly`

File: formula/src/test/java/com/instacart/formula/MultipleEffectTest.kt [hunk 6]
```diff
@@ -6,7 +6,7 @@ import com.instacart.formula.test.test
 import io.reactivex.Observable
 import org.junit.Test
 
-class MultipleMessageTest {
+class MultipleEffectTest {
 
     @Test fun `multiple message order should be maintained`() {
```
Comment (Jawnnypoo) [inline formula/src/test/java/com/instacart/formula/MultipleEffectTest.kt]: `multiple effects order should be maintained`

**Pull request merged**